base/Enums.jl

```diff
@@ -140,4 +141,150 @@ macro enum(T,syms...)
     return blk
 end
 
+
+
+abstract FlagEnum <: AbstractEnum
+
+Base.convert{T<:Integer}(::Type{T}, x::FlagEnum) = convert(T, convert(Unsigned, x))
+for op in (:|, :&, :xor)
+    @eval function Base.$op{T<:FlagEnum}(x::T,y::T)
+        reinterpret(T, ($op)(convert(Unsigned, x), convert(Unsigned, y)))
+    end
+end
+Base.in{T<:FlagEnum}(x::T, y::T) = x & y == x
+Base.~{T<:FlagEnum}(x::T) = reinterpret(T, convert(Unsigned, typemax(T)) & ~convert(Unsigned, x))
+
+
+"""
+    @enumset EnumName[::U] EnumValue1[=x] EnumValue2[=y]
+
+Create an [`FlagEnum`](:obj:`FlagEnum`) type with name `EnumName` and base member values of
+`EnumValue1` and `EnumValue2`, based on the unsigned integer type `U` (`UInt32` by
+default).  The optional assigned values of `x` and `y` must have exactly 1 bit set, and
+not overlap. `EnumName` can be used just like other types and enum member values as
+regular values, such as
+
+```jldoctest
+julia> @enumset FRUITSET apple=1<<0 orange=1<<1 kiwi=1<<2
```

Comment:
`@flagenum` here as well, and maybe rename FRUITSET to something else (FRUITFLAGS?)